### ISSUE TITLE
Cannot install ZLED-RGB9 and ZLED-G2705

### ISSUE THREAD

**Issue by bm480** [athombv/nl.klikaanklikuit.zigbee#3]
I have a lot of Wave devices attached to Homey (with different rate of success) and I'm in for something completely different. So I bought a dew ZigBee devices from KlikAanKlikUit to find out whether or not it is an improvement or not.

However, I'm unable to pair the RGB9 device. I've tried the following (20 times):
- Press (+) to add new ZigBee device to Homey
- Put the device in pair mode (switch off/on 5x), lamp turns green
- Homey displays: Verkrijg informatie over endpoint (2/2), cluster (0/0)...
- Homey waits for 10 to 20 seconds, lamp already turns white again
- Sometimes I can see cluster counting up to 9, sometimes it just continues without
- Homey briefly displays: Zoek naar geschikte app
- A new device is created: Onbekend ZigBee apparaat

The device info for unknown RGB9 device is:

ProductID ZLL-ExtendedColo
ApparaatID 4096
ProfielID 49246

Roughly the same applies to the ZLED-G2705. After pairing an unknown device is created.
The device info for unknown G2705 device is:

ProductID ZLL-DimmableLig
ApparaatID 256
ProfielID 49246

**Comment by bm480** [comment]
I've also tried resetting the ZigBee network a few times (since it is the first device), without any results.

**Comment by RobinBol** [comment]
Did you use the latest version of the kaku zigbee app? As this device is present in the drivers: 
```
ProductID ZLL-ExtendedColo
ApparaatID 4096
ProfielID 49246
```

And are you sure that the second device's productid = 'ZLL-DimmableLig' and not 'ZLL-DimmableLigh' as the latter is also present in the drivers.

**Comment by bm480** [comment]
I'm very ashamed to notice that I forgot to install the Klik Aan Klik Uit Zigbee app...

I just installed it but still did not manage to pair the device within a number of retries. At least I'm not getting the unknown device anymore. However, I will try again later today and let you know if it worked out.

**Comment by RobinBol** [comment]
No problem, let me know if it works out. Tip: follow the instructions in the KaKu pairing windows to reset the bulbs before pairing.

**Comment by bm480** [comment]
I finally managed to pair the ZLED-RGB9 device. I had to reset the ZigBee network a few times though. The ZLED-RGB9 is put into pairing mode by switching the bulb off and on 5 times in a row. Sometimes Homey is already starting the pairing process after the first or second switch while the bulb itself is not in pairing mode at all.

At least I'm now able to test the device. I have a simple flow that turns on the light and sets color temperature and brightness to the maximum level. Which immediately shows one of the advantages of this device, since the light it produces is warm, bright and its easy to find the right colors for different moods. However, I found that the maximum distance from Homey is somewhat limited. I have a few Zipato Z-Wave bulbs which can be put anywhere in the house (which I admit has about 75 other Z-Wave devices some of which are capable of routing the signal). In my L-shaped room when Homey is in the upper-left part of the L and the bulb is in the lower-right part of the L it is not possible to control the device. Only when placing the device within direct line of sight, I was able to have Homey control the device. That being said, sometimes the device reacts terrible slow (takes more than 5 seconds) and sometimes nothing happens at all. But this could be due to the immaturity  of the ZigBee implementation of Homey which is still in Beta. Slow response times also happen with my Z-Wave devices, so it could also be caused by the Homey software itself (1.5.2).

I will try to pair the ZLED-G2705 tomorrow and try to find out if this improves the range at which devices can be controlled a bit. I'm also curious about how fast these devices respond when they are turned on or off at the same time. With Z-Wave or non-ZigBee KlikAanKlikUit I found out it is not possible to control many lights in such way that they are turned on or off at almost the same time. You always experience the Christmas tree effect where lights change one device at a time over a period of tens of seconds.

**Comment by RobinBol** [comment]
1. ZigBee devices do not have a pairing mode such as Z-Wave devices, when a ZigBee device is reset it will broadcast a message when it is turned on. Which explains why it sometimes started pairing before it entered 'pairing mode'.

2. ZigBee range might vary and is not comparable with your Z-Wave mesh. Btw, your Z-wave devices responding slow is not very strange when you have approximately 75 Z-Wave devices in your network.

I will close this issue for now as you managed to pair the device, if you have other problems please create separate issues for that preferably in this repo (https://github.com/athombv/homey/issues) when they are not directly related to this app. Thanks!

**Comment by bm480** [comment]
At last I managed to pair both devices with Homey. One odd thing to mention: if you put Homey into ZigBee pairing mode and just turn on the new unpaired device only once, Homey is still able to pair the device. If you try to put the device into pairing mode (as per instructions by turning it off and on 5 times) Homey gets confused and quickly responds with a vague error message.

Now for the range and performance. The G2705 was placed at about 3 meters from Homey, while the RGB9 was at about 11 meters from the G2705 and not within direct vicinity of Homey (remember the L-shaped room?) I made a test-flow to switch both devices on after a second, then switch them off a second later. At first it seemed that both devices responded fast and almost simultaneously in both cases. I was surprised that the RGB9 responded better than before and thought the improvement must have been caused by the G2705 acting as a router. However, I then tried to even broaden the range and position the RGB9 in the far corner so that it was even more difficult for Homey to reach it directly. But also not in direct line of sight of the RGB9. And indeed, the RGB9 did not respond to Homey anymore (what could of course be expected). Then I positioned the RGB9 in direct vicinity of the G2705 but more than 11 meters away. The RGB9 sometimes responded but many times it didn't in which case the actions in the test flow indicated the following error: AF data request fails, status code: 233. Mac no ack.

Then I looked at the ZigBee settings, the Stuff for Geeks panel. And to my surprise I saw both devices directly connected to Homey but not to each other. The G2705 had a percentage of 15% while the RGB9 had a percentage of 0% (no idea what it means). When I positioned the RGB9 near the G2705 and also at 3 meters from Homey communication was a lot smoother. Percentages for connections wit Homey at 11% and 17% and connection between both devices at 32%. I must admit that the circumstances were not ideal, the RGB9 was put in a metal lamp which might have impact on communications. But still, communication between devices in the same room should not be a problem at all. Finally I put the RGB9 in its initial position, and repeated the test. Performance was adequate and I was surprised to see that the G2705 did act as a router (at least according to the Geekstuff panel). But after a few minutes the routing graph reverted to the situation in which I first started.

At this point I'm not quite satisfied with both products. Communications with the RGB9 being very unreliable still and the G2705 not acting as a router. Also, the G2705 having a white colored housing did not quite match my metalic lamp housing. I'm not sure if I should invest in ZigBee devices... Any help and/or tips are very much appreciated.
